src/Features/CSharp/Portable/Completion/CompletionProviders/ObjectAndWithInitializerCompletionProvider.cs

```diff
@@ -144,6 +145,18 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
                 return Tuple.Create(type, token.GetLocation());
             }
 
+            // new() { bar = $$
+            if (token.Parent.Parent.IsKind(SyntaxKind.ImplicitObjectCreationExpression, out ImplicitObjectCreationExpressionSyntax implicitObjectCreation))
+            {
+                var type = semanticModel.GetTypeInfo(implicitObjectCreation, cancellationToken).ConvertedType;
+                if (type is ITypeParameterSymbol typeParameterSymbol)
+                {
+                    return Tuple.Create<ITypeSymbol, Location>(typeParameterSymbol.GetNamedTypeSymbolConstraint(), token.GetLocation());
+                }
```

Comment:
this feels duplicative.  i'd prefer we jsut return the type, and have a higher layer say what to do in teh case of a type-parameter.  (this applies to the ObjCreation case above as well).

Reply:
thanks! :)